dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1106,38 +1107,60 @@ private void provisionWorkflowVersionUploadFiles(ApiClient zendoClient, Deposit
         }
 
         // Add workflow version source files as a zip to the DOI upload deposit
-        // Borrow code from getWorkflowZip
         Set<SourceFile> sourceFiles = workflowVersion.getSourceFiles();
-        java.nio.file.Path path = Paths.get(workflowVersion.getWorkingDirectory());
         if (sourceFiles == null || sourceFiles.size() == 0) {
             LOG.warn("No source files found to zip when creating DOI");
             throw new CustomWebApplicationException("No source files found to"
                     + " upload when creating DOI. Zenodo requires at lease one file"
                     + " to be uploaded in order to create a DOI.", HttpStatus.SC_BAD_REQUEST);
         } else {
-            String fileName = workflow.getWorkflowPath().replaceAll("/", "-") + ".zip";
             OutputStream outputStream;
+            String versionOfWorkflow = workflowVersion.getName();
+
+            String fileNameBase = workflow.getWorkflowPath().replaceAll("/", "-")
+                    + "_" + versionOfWorkflow;
+            String fileSuffix = ".zip";
+            String fileName = fileNameBase + fileSuffix;
+            java.nio.file.Path tempDirPath;
+            try {
+                tempDirPath = Files.createTempDirectory(null);
+            } catch (IOException e) {
+                LOG.error("Could not create Zenodo temp upload directory." + " Error is " + e.getMessage(), e);
+                throw new CustomWebApplicationException("Could not create Zenodo upload temp directory"
```

Comment:
I would not give that level of detail to the client, and would instead make it more generic.

Reply:
https://ucsc-cgl.atlassian.net/browse/DOCK-861